server/protocols/webadmin-integration-test/distributed-webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/rabbitmq/RabbitMQEventDeadLettersIntegrationTest.java

```diff
@@ -139,42 +145,73 @@ void callsPerEventBeforeSuccess(int retriesBeforeSuccess) {
         private RetryEventsListener retryEventsListener;
+        private RetryEventsListener2 retryEventsListener2;
 
         @Override
         public void beforeEach(ExtensionContext extensionContext) throws Exception {
             retryEventsListener = new RetryEventsListener();
+            retryEventsListener2 = new RetryEventsListener2();
         }
 
         @Override
         public Module getModule() {
-            return binder -> Multibinder.newSetBinder(binder, MailboxListener.GroupMailboxListener.class)
-                .addBinding()
-                .toInstance(retryEventsListener);
+            return binder -> {
+                Multibinder<MailboxListener.GroupMailboxListener> setBinder = Multibinder.newSetBinder(binder, MailboxListener.GroupMailboxListener.class);
+                setBinder.addBinding().toInstance(retryEventsListener);
+                setBinder.addBinding().toInstance(retryEventsListener2);
+            };
         }
 
         @Override
         public boolean supportsParameter(ParameterContext parameterContext, ExtensionContext extensionContext) throws ParameterResolutionException {
-            return parameterContext.getParameter().getType() == RetryEventsListener.class;
+            Class<?> paramType = parameterContext.getParameter().getType();
+            return paramType == RetryEventsListener.class
+                || paramType == RetryEventsListener2.class;
         }
 
         @Override
         public Object resolveParameter(ParameterContext parameterContext, ExtensionContext extensionContext) throws ParameterResolutionException {
-            return retryEventsListener;
+            Class<?> paramType = parameterContext.getParameter().getType();
+            if (paramType == RetryEventsListener.class) {
+                return retryEventsListener;
+            } else if (paramType == RetryEventsListener2.class) {
+                return retryEventsListener2;
+            }
+
+            throw new IllegalArgumentException("unsupported type");
         }
     }
-    
+
+    private static RabbitMQExtension RABBIT_MQ_EXTENSION = new RabbitMQExtension();
     @RegisterExtension
     static JamesServerExtension testExtension = new JamesServerBuilder()
         .extension(new DockerElasticSearchExtension())
         .extension(new CassandraExtension())
         .extension(new AwsS3BlobStoreExtension())
-        .extension(new RabbitMQExtension())
+        .extension(RABBIT_MQ_EXTENSION)
         .extension(new RetryEventsListenerExtension())
         .server(configuration -> GuiceJamesServer.forConfiguration(configuration)
             .combineWith(CassandraRabbitMQJamesServerMain.MODULES)
-            .overrideWith(new WebadminIntegrationTestModule()))
+            .overrideWith(new WebadminIntegrationTestModule())
+            .overrideWith(binder -> {
+                try {
+                    binder.bind(RabbitMQConnectionFactory.class)
```

Comment:
why do you need that?

Reply:
with the test case I added in the integration test, `RabbitMQEventBus.dispatch()` takes so long, I can not say exactly how long, I was waiting over 20 minutes but it doesn't finish. So I guest we're doing to much retries: 
 - at `RabbitMQConnectionFactory.connectionMono()` 8 retries
 - at `SimpleConnectionPool.getResilientConnection()` 100 retries
 - at `ReactorRabbitMQChannelPool.borrow()` 3 retries
 - at `ReactorRabbitMQChannelPool.ChannelFactory` 5 retries

To many retries

Reply:
it's unclear by reading the code that the problem is `dispatch` taking too long to recover.
Does it mean if Rabbit is unresponsive in production we will have the same recovery problem?

Reply:
That's related to @remk work, showing that eventBus recovery in some tests needs to be at least 30 seconds due to some hard coded values in the driver. I do not recall all the details. Maybe @remk would help ;-)

Obviously we don't want to be waiting 30 seconds in every tests that relies on a failing eventBus.

However in production recovery will be fast.If the utage is small, we recover from it quickly.

Reply:
I bet this can be decreased `SimpleConnectionPool.getResilientConnection() 100 retries`

I propose a downgrade to 5

Reply:
I also rework duc tran proposal (about configuration fields) to have it at the extension level, and in a more explicit way.